File: Neetcode/ArrsHashes/twosum.java
import java.util.HashMap;

class Solution {

    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hmm = new HashMap<Integer, Integer>();
        int temp = 0;
        for (int i = 0; i < nums.length; i++) {
            temp = target - nums[i];
            if (hmm.containsKey(temp)) {
                return new int[] { hmm.get(temp), i };
            }
            hmm.put(nums[i], i);
        }
        return new int[] { -1, -1 };
    }
}

File: Neetcode/ArrsHashes/twosum.py
# https://neetcode.io/problems/two-integer-sum
# https://leetcode.com/problems/two-sum/
class Solution(object):
    def twoSum(self, nums, target):
        search = {}
        for i, n in enumerate(nums):
            curr = target - n
            if curr in search:
                return search[curr], i
            else:
                search[n] = i
